indico/modules/categories/models/event_move_request.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
+from indico.util.date_time import now_utc
+from indico.util.enum import RichIntEnum
+from indico.util.locators import locator_property
+
+
+class MoveRequestState(RichIntEnum):
+    pending = 0
+    accepted = 1
+    rejected = 2
+    withdrawn = 3
+
+
+class EventMoveRequest(db.Model):
+    """A request that represents a proposed event move."""
+
+    __tablename__ = 'event_move_requests'
+    __table_args__ = (db.Index(None, 'event_id', unique=True,
+                               postgresql_where=db.text(f'state = {MoveRequestState.pending}')),
+                      db.CheckConstraint(f'state in ({MoveRequestState.accepted}, {MoveRequestState.rejected}) '
+                                         f'AND moderator_id IS NOT NULL OR moderator_id IS NULL',
```

Comment:
I really do not want AND and OR on the same level without parentheses... this just makes it much harder to see the logic (and honestly I never remember which one has priority).

Reply:
Also, the logic here looks wrong: It unconditionally allows a NULL moderator_id, but this should only be the case in the *other* states.

So the check condition should be `(state in (...)) = (moderator_id IS NOT NULL)`

Reply:
Added also in the migration file.